### PR TITLE
Added feature to bind on stream emits via options

### PR DESCRIPTION
**Pull request by shamasis**
Adds feature to:

request.defaults({
  bindOn: { request: console.log }
});

## PR Checklist:
- [ ] I have run `npm test` locally and all tests are passing.
- [ ] I have added/updated tests for any new behavior.
       
- [ ] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -149,7 +149,21 @@ Request.prototype.init = function (options) {
     options = {}
   }
   self.headers = self.headers ? copy(self.headers) : {}
-
+  
+  // bind default events sent via options
+  // additional postman feature starts
+  if (options.bindOn) {
```
Comment (kunagpal) [inline request.js]: Needs a stricter type check.

Comment (shamasis) [reply request.js]: I don't see request library doing it anywhere

**Comment by kunagpal** [comment]
@shamasis Tests failing due to https://github.com/request/request/issues/2976

**Comment by kunagpal** [comment]
Blocked on https://github.com/postmanlabs/postman-request/pull/19

**Comment by kunagpal** [comment]
TODO: Add tests before stable release (CodeCov should be happy)

**Pull request merged**